samples/GraphQL.DataLoader.Example/GraphQl/SalespeopleByNameDataLoader.cs

```diff
@@ -0,0 +1,24 @@
+﻿using DataLoaderGql.Types;
+using GraphQL.DataLoader;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataLoaderGql.GraphQl;
+
+public class SalespeopleByNameDataLoader(DealershipDbContext db) : DataLoaderBase<string, Salesperson>
+{
+    protected override async Task FetchAsync(IEnumerable<DataLoaderPair<string, Salesperson>> list, CancellationToken cancellationToken)
+    {
+        IQueryable<Salesperson> salesmen = db.Salespeople;
+        var names = list.Select(pair => pair.Key);
+
+        var lookup = await salesmen
+            .Where(sm => names.Contains(sm.Name))
+            .GroupBy(sm => sm.Name)
+            .ToDictionaryAsync(group => group.Key, cancellationToken).ConfigureAwait(false);
```

Comment:
Combining GroupBy and ToDictionary is the equivalent of ToLookup, and this could be written more like `.ToListAsync(...).ToLookup()` which would be a better representation of how the data is used.

We should also consider if the name is unique, in which case a dictionary without the groupby would be a better choice, eliminating Single() below and will provide a more streamlined example.

Reply:
Another option is to change how this is used to be a collection data loader, which just returns the lookup, so a name may return multiple entries.  I just feel like Single() is a bad database design, and would rather not include that in the sample, favoring a better design and/or simpler.

Reply:
I couldnt figure out how to make a Collection DataLoader like this work. But maybe i can transform this one into the other Pattern?

Reply:
The CarsBySalespersonDataLoader is an accurate representation of a collection data loader

Reply:
ah yes, i remember. I think this is why i looked to make this one a non-Collection One.
 I can make the Name a unique Index and FirstOrDefault as more usual Db Pattern. 
 Maybe we do a second little Project, that shares the overall Structure of this but uses the Other way of using Dataloaders, and Reference Both in the Docs.
 What do you think about that?
 I think we wouldnt want to recommend to mix them, as i think in the Docs it sais 'If u use this, you can leave the other stuff out' about the DI version

Reply:
Good idea

Reply:
Pushed it :) Also Referenced the DataLoader Docs. 
Those will ofcourse need to be a bit reworked and should then probably reference Code from the Sample Projects as far as possible, but i'd really prefer to do that in a seperate PR, after this one is closed. This one is already quite big and long running.